File: app/src/main/java/com/example/task2/javafolder/viewviewmodel/fragments/startingfragments/ParentStartingFragment.java
package com.example.task2.javafolder.viewviewmodel.fragments.startingfragments;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.navigation.Navigation;

import com.example.task2.databinding.FragmentStartingBinding;
import com.example.task2.javafolder.viewviewmodel.fragments.BaseFragment;
import com.example.task2.javafolder.viewviewmodel.fragments.parentfragment.ParentViewModel;
import com.google.android.material.textfield.TextInputEditText;

import dagger.hilt.android.AndroidEntryPoint;

public abstract class ParentStartingFragment extends BaseFragment {

    private FragmentStartingBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentStartingBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setCalculationsButtonClicker();
        binding.textView.setText(stringIdForTextView());
        binding.textField.setHint(stringIdForEditTextHint());
    }

    public abstract int determineNextFragment();

    @StringRes
    public abstract int stringIdForTextView();

    @StringRes
    public abstract int stringIdForEditTextHint();

    @Override
    public final TextInputEditText getUsersInputInstance() {
        return binding.usersInput;
    }

    @Override
    public Button getButtonInstance() {
        return binding.button;
    }

    private void setCalculationsButtonClicker() {
        binding.button.setOnClickListener(view -> Navigation.findNavController(view).navigate(determineNextFragment()));
    }
}

File: app/src/main/java/com/example/task2/kotlinfolder/viewviewmodel/fragments/startingfragments/ParentStartingFragment.kt
package com.example.task2.kotlinfolder.viewviewmodel.fragments.startingfragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.annotation.StringRes
import androidx.navigation.Navigation
import com.example.task2.databinding.FragmentStartingBinding
import com.example.task2.kotlinfolder.viewviewmodel.fragments.BaseFragment
import com.google.android.material.textfield.TextInputEditText

abstract class ParentStartingFragment :
    BaseFragment<FragmentStartingBinding, ParentStartingViewModel>() {

    override fun getViewBinding() = FragmentStartingBinding.inflate(layoutInflater)

    override fun initViews() {
        super.initViews()
        setCalculationsButtonClicker()
        binding.textView.setText(stringIdForTextView)
        binding.textField.setHint(stringIdForEditTextHint)
    }

    abstract val nextFragmentId: Int

    @get:StringRes
    abstract val stringIdForTextView: Int

    @get:StringRes
    abstract val stringIdForEditTextHint: Int

    override val usersInputEditText: TextInputEditText get() = binding.usersInput

    override val calculateButton: Button get() = binding.button

    private fun setCalculationsButtonClicker() {
        binding.button.setOnClickListener {
            Navigation.findNavController(requireView()).navigate(nextFragmentId)
        }
    }
}
